public/event-emitters/event-emitter-base.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Defines the event emitter API.
+ */
+
+var _ = require('lodash');
+var P = require('pjs').P;
+
+var ListenerWrapper = P({
+  init: function(callback, remove_after_call) {
+    this.callback = callback;
+    this.remove_after_call = remove_after_call;
+  }
+});
+
+var check_action_class = function(action_class) {
+  if (!action_class.prototype.key) {
+    console.error("Bad action class, should have a key", action_class);
```

Comment:
Also, should this `throw` instead of just log an error? Seems like a serious bug.